metl/model_encoder.py

```diff
@@ -0,0 +1,114 @@
+import torch
+from typing import Literal
+from biopandas.pdb import PandasPdb
+import metl.relative_attention as ra
+from Bio.SeqUtils import seq1
+import os
+
+class ModelEncoder(torch.nn.Module):
+    def __init__(self, model, encoder, strict=True, indexing:Literal[0,1] = 0) -> None:
+        super(ModelEncoder, self).__init__()
+
+        if indexing != 0 and indexing != 1:
+            raise Exception("Indexing must be equal to 0 or to 1.")
+
+        self.model = model
+        self.encoder = encoder
+
+        self.indexing = indexing
+        self.strict = strict
+
+        self.needs_pdb = self.check_if_pdb_needed(model)
+
+    def check_if_pdb_needed(self, model):
+        sequential = next(model.children())
+        
+        for layer in sequential:
+            if isinstance(layer, ra.RelativeTransformerEncoder):
+                return True
+        return False
+
+    def validate_pdb(self, pdb_file, wt):
+        try:
+            ppdb = PandasPdb().read_pdb(pdb_file)
+        except Exception as e:
+            raise Exception(f"{str(e)} \n\n PDB file could not be read by PandasPDB. It may be incorrectly formatted.")
+
+        groups = ppdb.df['ATOM'].groupby('residue_number')
+        wt_seq = []
+        for group_name, group_data in groups:
+            wt_seq.append(seq1(group_data.iloc[0]['residue_name']))
+        wildtype = ''.join(wt_seq)
+
+        if self.strict:
+            err_str = "Strict mode is on because a METL model that we trained was used. Wildtype and PDB sequeunces must match."
+            err_str += " If this is expected behavior, pass strict=False to the load function you used."
+            assert wildtype == wt, err_str
+
+    def validate_variants(self, variants, wt):
+        wt_len = len(wt)
+        for index, variant in enumerate(variants):
+            split = variant.split(',')
+            for mutation in split:
+                from_amino_acid = mutation[0]
+                to_amino_acid = mutation[-1]
+                location = int(mutation[1:-1])
+
+                errors = []
+
+                if location <= 0 or location >= wt_len-1:
```

Comment:
It wasn't obvious to me at first that this is after everything has been translated to 0-based indexing. We should comment that validation can only be run after conversion to 0-based.

Reply:
yea, that would be good. I'll add it